actors/builtin/power/power_state.go

```diff
@@ -146,23 +159,49 @@ func (st *State) addToClaim(s adt.Store, miner addr.Address, power abi.StoragePo
 		return errors.Errorf("no claim for actor %v", miner)
 	}
 
+	oldNominalPower, err := st.computeNominalPower(s, miner, claim.Power)
+	if err != nil {
+		return err
+	}
+
+	// update pledge and power
 	claim.Power = big.Add(claim.Power, power)
 	claim.Pledge = big.Add(claim.Pledge, pledge)
+
+	newNominalPower, err := st.computeNominalPower(s, miner, claim.Power)
+	if err != nil {
+		return err
+	}
+
+	if oldNominalPower.LessThan(ConsensusMinerMinPower) && newNominalPower.GreaterThanEqual(ConsensusMinerMinPower) {
+		// just passed min miner size
+		st.NumMinersMeetingMinPower++
+	} else if oldNominalPower.GreaterThanEqual(ConsensusMinerMinPower) && newNominalPower.LessThan(ConsensusMinerMinPower) {
+		// just went below min miner size
+		st.NumMinersMeetingMinPower--
+	}
+
 	AssertMsg(claim.Power.GreaterThanEqual(big.Zero()), "negative claimed power: %v", claim.Power)
 	AssertMsg(claim.Pledge.GreaterThanEqual(big.Zero()), "negative claimed pledge: %v", claim.Pledge)
+	AssertMsg(st.NumMinersMeetingMinPower >= 0, "negative number of miners larger than min: %v", st.NumMinersMeetingMinPower)
 	return st.setClaim(s, miner, claim)
 }
 
 func (st *State) computeNominalPower(s adt.Store, minerAddr addr.Address, claimedPower abi.StoragePower) (abi.StoragePower, error) {
 	// Compute nominal power: i.e., the power we infer the miner to have (based on the network's
 	// PoSt queries), which may not be the same as the claimed power.
-	// Currently, the only reason for these to differ is if the miner is in DetectedFault state
-	// from a SurprisePoSt challenge. TODO: hs update this
+	// Currently, the nominal power may differ from claimed power because of
+	// collateral and faults (declared or detected)
```

Comment:
It doesn't differ due to declared faults. Declared faults are already reflected in the claimed power - a miner is not claiming a faulty sector has power. The claim will already include any "trickiness".

This means the `else` case below is unnecessary (but maybe add a comment so someone else doesn't have the same misconception).

Reply:
ahhhh I had misunderstood your above comment, as you saw. Will update accordingly. With that said, could you tell me where `DeclareTempFaults` updates the "power `claim`s"? I see where it updates the sector itself in the miner actor, but not where it updates the `Claim` from the power actor. Is that missing (should be handled in subseq PR)?

Reply:
DeclareTempFaults registers a cron handler that does the real work. See `requestBeginFaults` which invokes the power actor.

Reply:
I was missing it: it all happens thanks to the cron actor in `checkTemporaryFaultEvents()`